CRM/Contribute/Form/Contribution.php

```diff
@@ -466,6 +466,12 @@ public function setDefaultValues() {
       $defaults['campaign_id'] = $this->_pledgeValues['campaign_id'];
     }
 
+    if (!empty($defaults['address_id'])) {
+      $addressParams = ['id' => CRM_Utils_Array::value('address_id', $defaults)];
+      $addressDetails = CRM_Core_BAO_Address::getValues($addressParams, FALSE, 'id');
+      $addressDetails = array_values($addressDetails);
+      $this->assign('billing_address', $addressDetails[0]['display']);
+    }
```

Comment:
Overall this PR looks good but some notes:

* We don't use CRM_Utils_Array::value() anymore if possible (use php 7 `??`), but also this is inside an `if` where it already checks if the value exists, so it's not necessary.
* You need to `$this->assign('billing_address')` outside the `if` so that it always has a value, since otherwise the tpl `{if $billing_address}` will give a warning. Maybe just before the `if` add `$this->assign('billing_address', NULL);` so it will be assigned _something_.

Reply:
@yashodha this just needs a minor tweak ^^^^ and then would be mergeable